tests/InstrEngineTests/InstrEngineTests/TargetAppCompiler.cs

```diff
@@ -53,18 +55,32 @@ internal class TargetAppCompiler
             "MultiReturnTests"
         };
 
-        internal static void ComplileCSharpTestCode(string path)
+        private const string DynamicCodeAssemblyName = "DynamicCodeAssembly";
+
+        internal static void ComplileCSharpTestCode(string directoryPath)
         {
+            EmitResult result;
             foreach (string prefix in TestAppFilePrefixes)
             {
-                string sourceCode = GetEmbeddedFile("InstrEngineTests.EmbeddedResources", prefix + ".cs");
+                string sourceCode = GetEmbeddedSourceFile(prefix);
                 SourceText sourceText = SourceText.From(sourceCode);
 
-                GenerateExecutable(sourceText, path, prefix, isDebug: true, is64bit: false);
-                GenerateExecutable(sourceText, path, prefix, isDebug: true, is64bit: true);
-                GenerateExecutable(sourceText, path, prefix, isDebug: false, is64bit: false);
-                GenerateExecutable(sourceText, path, prefix, isDebug: false, is64bit: true);
+                result = CompileTestAppPrefix(sourceText, directoryPath, prefix, isDebug: true, is64bit: false);
+                Assert.IsTrue(result.Success);
```

Comment:
Consider logging the errors from the compilation into the test result

Reply:
Done